pkg/virt-handler/vm.go

```diff
@@ -335,8 +335,14 @@ func (d *VirtualMachineController) execute(key string) error {
 		vmi.UID = domain.Spec.Metadata.KubeVirt.UID
 	}
 
-	// Ignore domains from an older VMI
+	// Ignore domains from an older VMI but delete watchdog file,
+	// by definition there can only be one VMI with the same name run at the same time.
+	// When we delete the watchdog file we trigger a DELETE for the damain cache for the stale entry
 	if vmiExists && domainExists && domain.Spec.Metadata.KubeVirt.UID != vmi.UID {
+		err := watchdog.WatchdogFileRemove(d.virtShareDir, vmi)
```

Comment:
I don't think we should be ignoring this. A stale watchdog is something we're meant to take notice of. It means the previous domain was not cleaned up properly. Deleting the watchdog file just makes the symptom go away.  When we encounter this, we need to ensure both the domain and all the domains artifacts are removed locally. That means more than removing the watchdog file. 

If the domain isn't running, a better option here would be to call processVmCleanup by passing in a fake VMI object that has the old Domains UID.